File: Server/source/lib/Json/Data/CBorrowHistory.php
<?php

namespace Json\Data;

class CBorrowHistory
{
	public $username;
	public $bookName;
	public $bookBianhao;
	public $borrowDate;
	public $planReturnDate;
	public $realReturnDate;
	public $ISBN;
		
	public function  __construct(\Models\BorrowHistory $borrowRecord)
	{
		$this->username = $borrowRecord->getUser()->getUsername();
		$this->bookName = $borrowRecord->getBook()->GetTitle();
		$this->bookBianhao = $borrowRecord->getBook()->GetBianHao();
		$this->ISBN = $borrowRecord->getBook()->getISBN();
		$this->borrowDate = $borrowRecord->getStartBorrowDate();
		$this->planReturnDate = $borrowRecord->getPlanReturnDate();
		$this->realReturnDate = $borrowRecord->getRealReturnDate();
	}
}

File: GTCCLibrary/GTCCLibrary/CBorrowHistory.m
//
//  CBorrowHistory.m
//  GTCCLibrary
//
//  Created by Lyle on 10/21/12.
//
//

#import "CBorrowHistory.h"


@implementation CBorrowHistory
@synthesize username,bookName,bookBianhao, borrowDate,planReturnDate,realReturnDate,ISBN;

-(void) Parse:(NSDictionary*) data
{
    self.username = [data objectForKey:@"username"];
    self.bookName = [data objectForKey:@"bookName"];
    self.bookBianhao = [data objectForKey:@"bookBianhao"];
    self.borrowDate = [data objectForKey:@"borrowDate"];
    self.planReturnDate = [data objectForKey:@"planReturnDate"];
    self.realReturnDate = [data objectForKey:@"realReturnDate"];
    self.ISBN = [data objectForKey:@"ISBN"];
}
@end
